articles/introduction-to-theory-of-computation/index.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: engineering-education
+status: publish
+published: 
+url: /engineering-education/introduction-to-theory-of-computation/
+title: Introduction to Theory of Computation
+description: This article aims to cover the basics of theory of computation and its role in efficiency of computers
+author: elly-omondi
+date: 2020-12-08T00:00:00-11:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/introduction-to-theory-of-computation/hero.jpg
+    alt: TOC cover image
+
+---
+
+
+This article will cover:
+
+- [Introduction](#Overview)
+- [Basics-of-Automata](#Automata-Basics)
+- [Computability](#Computability)
+- [Complexity](#Complexity)
+- [Conclusion](#Conclusion)
+
+
+### Introduction
+**[Theory of computation]**(https://en.wikipedia.org/wiki/Theory_of_computation) is a branch of Computer Science that is concerned with how problems can be solved  using algorithms and how efficiently they can be solved
+Real world computers perform computations that by nature run like mathematical models to solve problems in systematic ways. 
+The essence of theory of computation is to help develop mathematical and logical models that run efficiently and to a point of halting.Since all machines that implement logic apply TOC, studying TOC gives learners an insight of computer hardware and software limitations
+<!--more-->
+
+### Key considerations of computational problems
+1. What can and cannot be computed.
+2. Speed of such computations.
+3. The amount of memory in use during such computations.
+
+#### Importance of Theory of computation
+The theory of computation forms the basis for:
+* Writing efficient algorithms that run in computing devices
+* Programming language research and their development.
+* Efficient Compiler design and construction
+<!--more-->
+The Theory of Computation is made up of 3 branches i.e
+- Automata Theory
+- Computability Theory
+- Complexity Theory
+
+
+### Basic of Automata Theory
+This branch of theoretical computer science was developed by mathematicians to simplify the logic of computation by using well defined abstract computational devices(models).
+[Automata Theory](https://en.wikipedia.org/wiki/Automata_theory) is basically the study of abstract computational devices and forms a formal framework for design and analysis new types of computing devices such as bio computers and quantum computers.
```

Comment:
`design and analysis new types of computing devices`

after design and analysis, add `of`.